processor/processor_test.go

```diff
@@ -3412,7 +3409,11 @@ func assertDestinationTransform(
 				rawEvent, err := json.Marshal(event)
 				Expect(err).ToNot(HaveOccurred())
 				recordID := gjson.GetBytes(rawEvent, "message.recordId").Value()
-				Expect(event.Metadata.RecordID).To(Equal(recordID))
+				if recordID == nil {
```

Comment:
**Question:** did we add a scenario where recordID is now ni?

Reply:
We used to be getting empty string before but now we are getting nil

Reply:
Why are we getting nil? Is it a new scenario that we introduced?